x-pack/plugins/synthetics/server/routes/synthetics_service/assets_check.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { SyntheticsServerSetup } from '../../types';
+import { SyntheticsRestApiRouteFactory } from '../types';
+import { SYNTHETICS_API_URLS } from '../../../common/constants';
+
+const EXPECTED_PIPELINES = [
+  'synthetics-tcp',
+  'synthetics-http',
+  'synthetics-icmp',
+  'synthetics-browser',
+  'synthetics-browser.screenshot',
+  'synthetics-browser.network',
+];
+
+const EXPECTED_INDEX_TEMPLATES = ['synthetics', ...EXPECTED_PIPELINES];
+
+export const getSyntheticsAssetsCheckRoute: SyntheticsRestApiRouteFactory = () => ({
+  method: 'PUT',
+  path: SYNTHETICS_API_URLS.SYNTHETICS_ASSETS_CHECK,
+  validate: {},
+  writeAccess: true,
+  handler: async ({
+    server,
+  }): Promise<{ hasAllAssets: boolean; reinstalled?: boolean; error?: string }> => {
+    return await verifySyntheticsAssetsChecks(server);
+  },
+});
+
+export const verifySyntheticsAssetsChecks = async (server: SyntheticsServerSetup) => {
+  const esClient = server.coreStart.elasticsearch.client.asInternalUser;
+
+  const [pipelines, indexTemplates] = await Promise.all([
+    esClient.ingest.getPipeline({
+      id: 'synthetics-*',
+    }),
+    esClient.indices.getIndexTemplate({
+      name: 'synthetics*',
+    }),
+  ]);
+  const pipelineIds = Object.keys(pipelines);
+  const missingPipelines = EXPECTED_PIPELINES.filter(
+    (pipeline) => !pipelineIds.some((id) => id.startsWith(pipeline))
```

Comment:
`startsWith` is a bad candidate for this check.

If I delete `synthetics.browser` pipeline, as I did in my testing:

<img width="406" alt="image" src="https://github.com/elastic/kibana/assets/18429259/8798102c-5228-4486-bda0-d878ed301c95">

There are numerous entries in the remaining pipelines list that will `startsWith` the missing expected value. Even though `synthetics-browser-1.1.1` is missing from this list, `'synthetics-browser.screenshot-1.1.1'` and `'synthetics-browser.network-1.1.1'` both will satisfy the filter and we won't trigger the asset refresh.

One alternative could be to `split` on the key itself and perform a few assumption checks in the `some` predicate. Example:

```typescript
const missingPipelines = EXPECTED_PIPELINES.filter(
    (pipeline) => !pipelineIds.some((id) => {
      const segments = id.split(pipeline);
      return segments.length === 2 && segments[0] === '' && segments[1].startsWith('-');
  })
);
```

Example output:

```typescript
// successful match
'synthetics-browser.screenshot-1.1.1'.split('synthetics-browser.screenshot')
['', '-1.1.1']

// failed match
'synthetics-browser.screenshot-1.1.1'.split('synthetics-browser')
['', '.screenshot-1.1.1']
```

I'm open to any solution though so if you have a better option that's fine with me.